console-src/src/components/ClientForm/ClientForm.tsx

```diff
@@ -0,0 +1,208 @@
+    <CheckboxContainer>{children}</CheckboxContainer>
+  </Fragment>
+);
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: center;
+`;
+
+type ClientFormProps = {
+  action: Function;
+  isLoading?: boolean;
+  buttonLabel: string;
+  history: History;
+};
+
+type Scopes = {
+  ACCOUNTS: boolean;
+  USERS: boolean;
+  TRANSACTIONS: boolean;
+  TRANSFERS: boolean;
+  SUBSCRIPTIONS: boolean;
+  STATEMENTS: boolean;
+  [key: string]: boolean;
+};
+
+type State = {
+  isValid: boolean;
+  name?: string;
+  redirectUri?: string;
+  secret?: string;
+  scopes: Scopes;
+};
+
+type TextInputKeyType = "name" | "secret" | "redirectUri";
+
+const hasValidScopes = (scopes: Scopes) =>
+  Object.values(scopes).some(scope => scope);
+
+class ClientForm extends Component<ClientFormProps, State> {
+  constructor(props: ClientFormProps) {
+    super(props);
+
+    this.state = {
+      isValid: false,
+      name: undefined,
+      redirectUri: undefined,
+      secret: undefined,
+      scopes: {
+        ACCOUNTS: false,
+        USERS: false,
+        TRANSACTIONS: false,
+        TRANSFERS: false,
+        SUBSCRIPTIONS: false,
+        STATEMENTS: false
+      }
+    };
+  }
+
+  handleTextInputChange = (key: TextInputKeyType) => (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const { value } = event.target;
+    this.setState(state => {
+      const updatedState = {
+        ...state,
+        [key]: value
+      };
+      return {
+        ...updatedState,
+        isValid:
+          hasValidScopes(updatedState.scopes) && Boolean(updatedState.name)
+      };
+    });
+  };
+
+  handleCheckboxClick = (key: string) => () => {
+    this.setState((state: State) => {
+      const updatedScopes = {
+        ...state.scopes,
+        [key]: !state.scopes[key]
+      };
+      return {
+        ...state,
+        isValid: hasValidScopes(updatedScopes) && Boolean(state.name),
+        scopes: updatedScopes
+      };
+    });
+  };
+
+  submitForm = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const { name, redirectUri, secret, scopes } = this.state;
+
+    await this.props.action({
+      name,
+      redirectUri,
+      secret,
+      scopes: Object.keys(scopes).filter(scope => scopes[scope])
+    });
+
+    this.props.history.push("/clients");
+  };
+
+  render() {
+    const { name, redirectUri, secret, scopes, isValid } = this.state;
+    const { buttonLabel, isLoading } = this.props;
+
+    return (
+      <StyledForm onSubmit={this.submitForm}>
+        <TextInput
+          label={copy.clientForm.name}
+          placeholder={copy.clientForm.placeholders.name}
+          value={name}
+          handleChange={this.handleTextInputChange("name")}
+        />
+        <CheckboxGroup title={copy.clientForm.scopes.title}>
+          {Object.keys(scopes).map(scope => (
+            <Checkbox
+              // @ts-ignore
```

Comment:
I had some nasty error here and couldn't figure out a fix 🤔 if no better I'll leave this ignore for now

<img width="748" alt="Screenshot 2019-10-24 at 15 28 57" src="https://user-images.githubusercontent.com/3253522/67490157-11187c00-f673-11e9-9f4e-6d14c31a7b1a.png">

Reply:
Please see 744228a for a small suggestion.